pkg/api/handlers/serverinstances.go

```diff
@@ -0,0 +1,120 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	"github.com/obot-platform/obot/pkg/controller/handlers/accesscontrolrule"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ServerInstancesHandler struct {
+	acrHelper *accesscontrolrule.Helper
+	serverURL string
+}
+
+func NewServerInstancesHandler(acrHelper *accesscontrolrule.Helper, serverURL string) *ServerInstancesHandler {
+	return &ServerInstancesHandler{
+		acrHelper: acrHelper,
+		serverURL: serverURL,
+	}
+}
+
+func (h *ServerInstancesHandler) ListServerInstances(req api.Context) error {
+	var instances v1.MCPServerInstanceList
+	if err := req.List(&instances, kclient.MatchingFields{
+		"spec.userID": req.User.GetUID(),
+	}); err != nil {
+		return err
+	}
+
+	var convertedInstances []types.MCPServerInstance
+	for _, instance := range instances.Items {
+		convertedInstances = append(convertedInstances, convertMCPServerInstance(instance, h.serverURL))
+	}
+
+	return req.Write(types.MCPServerInstanceList{
+		Items: convertedInstances,
+	})
+}
+
+func (h *ServerInstancesHandler) GetServerInstance(req api.Context) error {
+	var instance v1.MCPServerInstance
+	if err := req.Get(&instance, req.PathValue("mcp_server_instance_id")); err != nil {
+		return err
+	}
+
+	return req.Write(convertMCPServerInstance(instance, h.serverURL))
+}
+
+func (h *ServerInstancesHandler) CreateServerInstance(req api.Context) error {
+	var input struct {
+		MCPServerID string `json:"mcpServerID"`
+	}
+	if err := req.Read(&input); err != nil {
+		return fmt.Errorf("failed to read server name: %w", err)
```

Comment:
Any time we fail to read input, we should return `types.NewErrBadRequest`.

Reply:
Thanks, I need to pay more attention to these things

Reply:
Fixed